File: primes1.py
#!/usr/bin/env python

def primes(limit):
    prime = [True] * (limit+1)
    prime[0:2] = False, False
    primes = []
    for (i, isprime) in enumerate(prime):
        if isprime:
            primes.append(i)
            for n in range(i*i, limit+1, i):
                prime[n] = False
    return primes

if __name__ == '__main__':
    import sys
    try:
        limit = int(sys.argv[1])
    except:
        print("{} LIMIT".format(sys.argv[0]))
    else:
        print(' '.join(map(str, primes(limit))))

File: primes1.swift
func primes(limit: Int) -> Int[] {
    var prime = Array(count: limit+1, repeatedValue: true)
    prime[0...1] = [false, false]
    var primes = Int[]()
    for (i, isprime) in enumerate(prime) {
        if isprime {
	    primes.append(i)
	    for var n=i*i; n<=limit; n+=i {
	        prime[n] = false
	    }
	}
    }
    return primes
}

func intifyarg1() -> Int? {
    if C_ARGC == 2 {
        let limitstr = String.fromCString(C_ARGV[1])
	if let limit = limitstr.toInt() {
	    return limit
        }
    }
    return nil
}

if let limit = intifyarg1() {
    let result = " ".join(map(primes(limit)) { 
        (var number) -> String in return number.description
    })
    println("\(result)")
} else {
    println("\(C_ARGV[0]) LIMIT")
}
